zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/repo/VFSNotebookRepo.java

```diff
@@ -238,14 +273,51 @@ public synchronized void save(Note note, AuthenticationInfo subject) throws IOEx
       throw new IOException(noteDir.getName().toString() + " is not a directory");
     }
 
-    FileObject noteJson = noteDir.resolveFile(".note.json", NameScope.CHILD);
+    FileObject noteJson = noteDir.resolveFile(".note.zpln", NameScope.CHILD);
+    
     // false means not appending. creates file if not exists
     OutputStream out = noteJson.getContent().getOutputStream(false);
     out.write(json.getBytes(conf.getString(ConfVars.ZEPPELIN_ENCODING)));
     out.close();
-    noteJson.moveTo(noteDir.resolveFile("note.json", NameScope.CHILD));
+    
+    setFileName(note, noteJson, noteDir);
```

Comment:
`.note.zpln` (noteJson) is temporary file where it's written initially, and further it's moved either into `note.json` if keeping old format, or `title.zpln` if saving in new format